### DISCUSSION TITLE
Error with reduce in CallPeaks function if no peaks are called

### DISCUSSION THREAD

**Discussion by zrcjessica** [stuart-lab/signac#614] [Ideas]
I am using `CallPeaks` to call cell type-specific peaks and have both `group.by` and `idents` arguments set. E.g.:

```R
DefaultAssay(dat) <- "peaks"
peaks <- CallPeaks(
    object = dat,
    group.by = "predicted.id",
    idents = celltype,
    macs2.path = "/iblm/netapp/home/jezhou/anaconda3/envs/r4/bin/macs2",
    outdir = file.path("test", celltype),
    name = celltype,
    additional.args = "--nomodel --shift 100 --extsize 200 --qval 5e-2 -B --SPMR",
    cleanup = FALSE)
```

In my example, no peaks were found for the cell type I was testing. Because the default value for `combine.peaks` is `TRUE`, the [function still attempted to call `reduce` on the GRanges object that results from the peak calling](https://github.com/timoast/signac/blob/2e4e18354297adc4b069b48a7794aa1efed35016/R/peaks.R#L137). This leads to an error message:

```
No peaks found for Cck+
Error in (function (classes, fdef, mtable)  : 
  unable to find an inherited method for function ‘reduce’ for signature ‘"NULL"’
Calls: CallPeaks -> CallPeaks.Seurat -> reduce -> <Anonymous>
Execution halted
```

This could be fixed by setting `combine.peaks` to `FALSE`; however, I wonder if it would be helpful for the function to automatically recognize when the GRanges object is empty and handle the error instead of crashing the job. 

Thanks!

**Comment by timoast** [comment]
Thanks for reporting, it would make sense to do a check on the number of peaks returned for each cell type before trying to combine them to avoid this error. We'll try to add this in a future release
